src/debug_wait.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#ifndef TIMESCALEDB_DEBUG_WAIT_H_
+#define TIMESCALEDB_DEBUG_WAIT_H_
+
+#include <postgres.h>
+
+#include <storage/lock.h>
+
+/* Debug waitpoint tag for debug waitpoints.
+ *
+ * Debug waitpoints only exists in debug-built code and is intended to allow
+ * more controlled testing of the code by creating wait-points where execution
+ * will halt until explicitly released.
+ *
+ * Each debug waitpoint is identified by a string that is hashed to a 8-byte
+ * number and used with the normal advisory locks available in PostgreSQL.
+ *
+ * When blocking on a waitpoint, there is an attempt to take a shared lock on
+ * the waitpoint. If the waitpoint is enabled by locking using an exclusive
+ * lock, this will block all waiters. Once the exclusive lock is released, all
+ * waiters will be able to proceed.
+ */
+typedef struct DebugWait
+{
+	const char *tagname;
+	LOCKTAG tag;
+} DebugWait;
+
+void debug_waitpoint_enable(DebugWait *waitpoint);
```

Comment:
I looks like only _init and _wait need to be exposed in this header file. The convention is also to prefix all public functions in non-TSL code with the `ts_` prefix. I am actually surprised it passes formatting checks without this prefix because we used to have a check for this: `scripts/export_prefix_check.sh.in`